src/Cli/dotnet/Commands/Package/Add/PackageAddCommandParser.cs

```diff
@@ -8,15 +8,50 @@
 using NuGet.Versioning;
 using Microsoft.DotNet.Tools.Package.Add;
 using Microsoft.DotNet.Cli.Extensions;
+using System.CommandLine.Parsing;
+using NuGet.Packaging.Core;
 
 namespace Microsoft.DotNet.Cli;
 
 internal static class PackageAddCommandParser
 {
-    public static readonly CliArgument<string> CmdPackageArgument = new DynamicArgument<string>(LocalizableStrings.CmdPackage)
+    public static NuGet.Packaging.Core.PackageIdentity ParsePackageIdentity(ArgumentResult packageArgResult)
+    {
+        // per the Arity of the CmdPackageArgument's Arity, we should have exactly one token - 
+        // this is a safety net for if we change the Arity in the future and forget to update this parser.
+        if (packageArgResult.Tokens.Count != 1)
+        {
+            throw new ArgumentException($"Expected exactly one token, but got {packageArgResult.Tokens.Count}.");
+        }
+        var token = packageArgResult.Tokens[0].Value;
+        var indexOfAt = token.IndexOf('@');
+        if (indexOfAt == -1)
+        {
+            // no version specified, so we just return the package id
+            return new NuGet.Packaging.Core.PackageIdentity(token, null);
+        }
+        // we have a version specified, so we need to split the token into id and version
+        else
+        {
+            var id = token[0..indexOfAt];
+            var versionString = token[(indexOfAt + 1)..];
+            if (NuGet.Versioning.SemanticVersion.TryParse(versionString, out var version))
+            {
+                return new NuGet.Packaging.Core.PackageIdentity(id, new NuGetVersion(version.Major, version.Minor, version.Patch, version.ReleaseLabels, version.Metadata));
+            }
+            else
+            {
+                throw new ArgumentException($"Invalid version string: {versionString}.");
```

Comment:
Localize?

Reply:
yeah, definitely need to pull that out. same with below.